spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/actions/AnalyzeTableSparkAction.java

```diff
@@ -63,97 +78,98 @@ protected AnalyzeTableSparkAction self() {
 
   @Override
   public Result execute() {
-    if (snapshotId == null) {
-      snapshotId = table.currentSnapshot().snapshotId();
-    }
     String desc = String.format("Analyzing table %s for snapshot id %s", table.name(), snapshotId);
     JobGroupInfo info = newJobGroupInfo("ANALYZE-TABLE", desc);
     return withJobGroupInfo(info, this::doExecute);
   }
 
   private Result doExecute() {
-    LOG.info("Starting the analysis of {} for snapshot {}", table.name(), snapshotId);
-    List<AnalysisResult> analysisResults =
-        types.stream()
-            .map(
-                statsName -> {
-                  switch (statsName) {
+    LOG.info("Starting analysis of {} for snapshot {}", table.name(), snapshotId);
+    List<AnalysisResult> results = Lists.newArrayList();
+    List<Blob> blobs =
+        blobTypesToAnalyze.stream()
+            .flatMap(
+                type -> {
+                  switch (type) {
                     case StandardBlobTypes.APACHE_DATASKETCHES_THETA_V1:
-                      return generateNDVAndCommit();
+                      try {
+                        return generateNDVBlobs().stream();
+                      } catch (Exception e) {
+                        LOG.error(
+                            "Error occurred when collecting statistics for blob type {}", type, e);
+                        ImmutableAnalyzeTable.AnalysisResult result =
+                            ImmutableAnalyzeTable.AnalysisResult.builder()
+                                .type(type)
+                                .addErrors(e.getMessage())
+                                .build();
+                        results.add(result);
+                      }
+                      break;
                     default:
-                      return ImmutableAnalyzeTable.AnalysisResult.builder()
-                          .type(statsName)
-                          .addAllErrors(Lists.newArrayList("Stats type not supported"))
-                          .build();
+                      throw new UnsupportedOperationException();
```

Comment:
In case this exception is thrown (due to some code modifications in the future), it could be helpful to include `type` in the exception message.